src/features/userprofile/models/UserProfileTypes.ts

```diff
@@ -7,10 +7,18 @@ export type UserProfileType = {
   readonly dateOfBirth?: Date;
   readonly language: string;
   readonly coverPic?: string;
+};
+
+export type UserProfileInfoType = UserProfileType & {
   readonly friends: ShortUserInfoType[];
 };
 
 export type ShortUserInfoType = Pick<
   UserProfileType,
   "id" | "firstName" | "lastName" | "profilePic"
 >;
+
+export type UpdateUserProfileProps = {
+  userId: string;
```

Comment:
I feel like we don't need this type -- only authenticated user can update their profile, and we can grab that userId from req.user, since we use jwt payload

Reply:
yeah or the body itself actually has the user.id field. I fixed it